packages/hop-node/src/watchers/SyncWatcher.ts

```diff
@@ -1686,6 +1742,53 @@ class SyncWatcher extends BaseWatcher {
       return true
     }
   }
+
+  // Experimental: Websocket support methods
+
+  initWebsocket = (contract: Contract, filter: EventFilter, cb: Function) => {
+    contract.on(filter, async (...event: any) => cb(event[event.length - 1]))
```

Comment:
I see here it's getting the last parameter as the event; for my understanding, what are the other parameters returned in the listener?

Reply:
`event` is the Event type generated by TypeChain. The interface and a practical example are below. There is likely a better way to handle it but I've never nailed down the best way to work with TypeChain so I did this. Happy to do something else if there is a more natural solution.

```

export interface TypedEventFilter<_TEvent extends TypedEvent>
  extends EventFilter {}


export interface TransferSentEventObject {
  transferId: string;
  chainId: BigNumber;
  recipient: string;
  amount: BigNumber;
  transferNonce: string;
  bonderFee: BigNumber;
  index: BigNumber;
  amountOutMin: BigNumber;
  deadline: BigNumber;
}
export type TransferSentEvent = TypedEvent<
  [
    string,
    BigNumber,
    string,
    BigNumber,
    string,
    BigNumber,
    BigNumber,
    BigNumber,
    BigNumber
  ],
  TransferSentEventObject
>;

export type TransferSentEventFilter = TypedEventFilter<TransferSentEvent>;
```

```
[
  '0xce6b702d315c03950030e5c2b008a2d8dab1d0af36c870cac76270a9e354a6ce',
  BigNumber { _hex: '0x05', _isBigNumber: true },
  '0xfEfeC7D3EB14a004029D278393e6AB8B46fb4FCa',
  BigNumber { _hex: '0x02c5ddeb108cfe52', _isBigNumber: true },
  '0x08dbc1e50017cd291b53a1e6ecea166ae6a2912e1e177b5020bf3c2e38bfd501',
  BigNumber { _hex: '0x237e4bc0f3a5e3', _isBigNumber: true },
  BigNumber { _hex: '0x01', _isBigNumber: true },
  BigNumber { _hex: '0x00', _isBigNumber: true },
  BigNumber { _hex: '0x00', _isBigNumber: true },
  {
    blockNumber: 15250984,
    blockHash: '0xfcea3a54861866acc699a5de406975b755f1f1a1d2ae17a4293f439756c2c34a',
    transactionIndex: 1,
    removed: false,
    address: '0x2708E5C7087d4C6D295c8B58b2D452c360D505C7',
    data: '0x00000000000000000000000000000000000000000000000002c5ddeb108cfe5208dbc1e50017cd291b53a1e6ecea166ae6a2912e1e177b5020bf3c2e38bfd50100000000000000000000000000000000000000000000000000237e4bc0f3a5e3000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000',
    topics: [
      '0xe35dddd4ea75d7e9b3fe93af4f4e40e778c3da4074c9d93e7c6536f1e803c1eb',
      '0xce6b702d315c03950030e5c2b008a2d8dab1d0af36c870cac76270a9e354a6ce',
      '0x0000000000000000000000000000000000000000000000000000000000000005',
      '0x000000000000000000000000fefec7d3eb14a004029d278393e6ab8b46fb4fca'
    ],
    transactionHash: '0x27aa876d282d5018363839988f2b753b87d8725f9987f025728a2f156c08438d',
    logIndex: 6,
    removeListener: [Function (anonymous)],
    getBlock: [Function (anonymous)],
    getTransaction: [Function (anonymous)],
    getTransactionReceipt: [Function (anonymous)],
    event: 'TransferSent',
    eventSignature: 'TransferSent(bytes32,uint256,address,uint256,bytes32,uint256,uint256,uint256,uint256)',
    decode: [Function (anonymous)],
    args: [
      '0xce6b702d315c03950030e5c2b008a2d8dab1d0af36c870cac76270a9e354a6ce',
      [BigNumber],
      '0xfEfeC7D3EB14a004029D278393e6AB8B46fb4FCa',
      [BigNumber],
      '0x08dbc1e50017cd291b53a1e6ecea166ae6a2912e1e177b5020bf3c2e38bfd501',
      [BigNumber],
      [BigNumber],
      [BigNumber],
      [BigNumber],
      transferId: '0xce6b702d315c03950030e5c2b008a2d8dab1d0af36c870cac76270a9e354a6ce',
      chainId: [BigNumber],
      recipient: '0xfEfeC7D3EB14a004029D278393e6AB8B46fb4FCa',
      amount: [BigNumber],
      transferNonce: '0x08dbc1e50017cd291b53a1e6ecea166ae6a2912e1e177b5020bf3c2e38bfd501',
      bonderFee: [BigNumber],
      index: [BigNumber],
      amountOutMin: [BigNumber],
      deadline: [BigNumber]
    ]
  }
]
```

Reply:
I see, seems fine the way it is 👍